src/LieTheory/WeylGroup.jl

```diff
@@ -753,6 +753,45 @@ function reflection(beta::RootSpaceElem)
   return W([list_of_indices; current_index; reverse(list_of_indices)])
 end
 
+###############################################################################
+# G-set functionality
+
+gset(W::WeylGroup) = gset(W, *, roots(root_system(W)); closed=is_finite(W))
```

Comment:
If `W` is infinite, I think `roots(root_system(W))` would be an infinite set ... ;-). But it doesn't hurt to do this.


But more importantly I wonder if we should really do this -- if you ask me "what is *the* $G$-set of a Weyl group", I am not sure I'd answer "well the one given by its action on its set of roots".

I realize this is probably here because we also have `gset(symmetric_group(5))` -- I think that one is not documented, and I also kinda think it was a mistake.

I think I'd prefer to call this, say, `natural_gset` (and document it accordingly). For backwards compatibility we should keep providing `gset(::PermGroup)`, but perhaps we can `@deprecate` it.

Thoughts, @lgoettgens @ThomasBreuer ?

Reply:
(to be clear, for this PR this would simply amount to
```suggestion
natural_gset(W::WeylGroup) = gset(W, *, roots(root_system(W)); closed=is_finite(W))
```
perhaps with a docstring (as one should probably explain and define what the "natural" G-set is for each type of group).


In addition or instead, how about supporting `gset(W, root_system(W))` ? That feels like pretty natural notation to me and it could be implemented more or less with the above code for `gset(::WeylGroup)`

Reply:
In order to drive things, perhaps for now just delete this:
```suggestion
```
and we can then discuss and decide whether to do `natural_gset`, or `gset(W, root_system(W))`, or both, or neither, after this PR is merged

Reply:
Not committing to this in this PR sounds good to me.
One comment: `roots(R)` will currently return the minimal roots of the root system. We have a large red box in the docs somewhere stating that current conventions and docstrings may only be sane for finite types and that there may be changes for infinite root systems